### PR TITLE
[Asset Inventory] Organization Support

### PR DESCRIPTION
**Pull request by kubasobon**
## Proposed commit message

The PR:
- Adds support for organization-wide deployment in AWS and Azure
- Reworks deployment type selection to use a drop-down menu
- Adds descriptions to credential fields, highlighting which ones to fill

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [x] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Related issues

Fixes https://github.com/elastic/security-team/issues/10606

## Screenshots

Screenshot 2024-10-14 at 10 05 31

_Screenshot showing the new deployment type drop-down and credential field descriptions_

### CODE REVIEW

File: packages/cloud_asset_inventory/data_stream/asset_inventory/manifest.yml [hunk 1]
```diff
@@ -15,115 +15,161 @@ streams:
...
         required: false
-        show_user: true
+        show_user: false
       - name: azure.credentials.client_password
         type: password
         title: Client Password
         multi: false
         required: false
-        show_user: true
+        show_user: false
```
Comment (romulets) [inline packages/cloud_asset_inventory/data_stream/asset_inventory/manifest.yml:156]: Why show_user for client username and client password became false?

Comment (kubasobon) [reply packages/cloud_asset_inventory/data_stream/asset_inventory/manifest.yml:156]: Required fields match those in CSPM. Username & Password are never used.

Screenshot 2024-10-17 at 10 17 02

**Comment by kubasobon** [comment]
@romulets Please take another look

**Review by romulets — APPROVED** [review]

**Comment by kubasobon** [comment]
Deferred until https://github.com/elastic/integrations/pull/11481 is merged

**Comment by kubasobon** [comment]
Revert https://github.com/elastic/cloudbeat/pull/2636 once this is merged.

**Pull request merged**